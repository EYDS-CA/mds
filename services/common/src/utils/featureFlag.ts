import { detectProdEnvironment as IN_PROD } from "./environmentUtils";
import flagsmith from "flagsmith";

// Name of feature flags. These correspond to feature flags defined in flagsmith.
export enum Feature {
  MAJOR_PROJECT_ARCHIVE_FILE = "major_project_archive_file",
  DOCUMENTS_REPLACE_FILE = "major_project_replace_file",
  MAJOR_PROJECT_ALL_DOCUMENTS = "major_project_all_documents",
  MAJOR_PROJECT_DECISION_PACKAGE = "major_project_decision_package",
  FLAGSMITH = "flagsmith",
  TSF_V2 = "tsf_v2",
}

// Definition of legacy flags
// TODO: Remove after flagsmith is live
const Flags = {
  [Feature.MAJOR_PROJECT_ARCHIVE_FILE]: !IN_PROD(),
  [Feature.DOCUMENTS_REPLACE_FILE]: !IN_PROD(),
  [Feature.MAJOR_PROJECT_ALL_DOCUMENTS]: !IN_PROD(),
  [Feature.MAJOR_PROJECT_DECISION_PACKAGE]: !IN_PROD(),
  [Feature.TSF_V2]: !IN_PROD(),
  [Feature.FLAGSMITH]: !IN_PROD(),
};

export const initializeFlagsmith = async (flagsmithUrl, flagsmithKey) => {
  await flagsmith.init({
    api: flagsmithUrl,
    environmentID: flagsmithKey,
    cacheFlags: true,
    enableAnalytics: true,
  });
};

/**
 * Returns true if the given feature is enabled
 * @param feature Feature to verify
 * @returns true if the given feature is enabled
 */
export const isFeatureEnabled = (feature: Feature) => {
  if (Feature.FLAGSMITH in Flags && Flags[Feature.FLAGSMITH]) {
    return flagsmith.hasFeature(feature);
  }

  if (feature in Flags) {
    return Flags[feature];
  }

  return false;
};
