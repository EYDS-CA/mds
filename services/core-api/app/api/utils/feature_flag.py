from enum import Enum

from app.config import Config
from flagsmith import Flagsmith
from flask import current_app


class Feature(Enum):
    TSF_V2 = 'tsf_v2'
    MAJOR_PROJECT_REPLACE_FILE = 'major_project_replace_file'
    MINE_APPLICATION_FILE_UDPATE_ALERTS = 'mine_application_file_update_alerts'
    TRACTION_VERIFIABLE_CREDENTIALS = 'verifiable_credentials'
    #if enabled the credential offer will be the current development state of all the 2.0 changes, Q1 2024
    VC_ANONCREDS_20 = 'vc_mines_act_permit_20'   # pending anoncred content differences.
    VC_ANONCREDS_CORE = "vc_anoncreds_core"
    VC_ANONCREDS_MINESPACE = "vc_anoncreds_minespace"
    VC_W3C = "vc_w3c"
    CODE_REQUIRED_REPORTS = 'code_required-reports'
    PERMIT_DOCUMENT_KEYWORD_SEARCH = 'permit_document_keyword_search'
    AMS_AGENT = 'ams_agent'

    def __str__(self):
        return self.value


flagsmith = Flagsmith(
    environment_key=Config.FLAGSMITH_KEY,
    api_url=Config.FLAGSMITH_URL,
)


def is_feature_enabled(feature):
    try:
        feature = str(feature).strip()
        flags = flagsmith.get_environment_flags()

        return feature in flags.flags and flags.is_feature_enabled(feature)
    except Exception as e:
        current_app.logger.error(f'Failed to look up feature flag for: {feature}. ' + str(e))
        return False
