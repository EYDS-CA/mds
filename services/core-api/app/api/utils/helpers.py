import re


def clean_HTML_string(raw_html):
    cleanr = re.compile('<.*?>')
    cleantext = re.sub(cleanr, '', raw_html)
    return cleantext


"""
Add sufix to permit number adding one to versioning.
"""


def generate_permit_no_sufix(permit, separator, filling=2):
    result = permit.split(separator + "-")
    version_str = '0' if len(result) < 2 else result[1]
    version = str(int(version_str) + 1).zfill(filling)
    permit_no = separator + "-" + version

    return permit_no
