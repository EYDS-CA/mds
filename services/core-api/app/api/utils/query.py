from flask_sqlalchemy import BaseQuery
from app.extensions import db


class QueryWithSoftDelete(BaseQuery):

    def __new__(cls, *args, **kwargs):
        obj = super(QueryWithSoftDelete, cls).__new__(cls)
        with_deleted = kwargs.pop('_with_deleted', False)
        if len(args) > 0:
            super(QueryWithSoftDelete, obj).__init__(*args, **kwargs)
            return obj.filter_by(deleted_ind=False) if not with_deleted else obj
        return obj

    def __init__(self, *args, **kwargs):
        pass

    def with_deleted(self):
        return self.__class__(
            db.class_mapper(self._entity_from_pre_ent_zero().class_),
            session=db.session(),
            _with_deleted=True)
