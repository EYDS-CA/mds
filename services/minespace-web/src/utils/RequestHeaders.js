import keycloak from "@/keycloak";
/**
 * Utility class for various request Headers.
 */

export const createRequestHeader = () => ({
  headers: {
    "Access-Control-Allow-Origin": "*",
    Authorization: `Bearer ${keycloak.token}`,
  },
});
