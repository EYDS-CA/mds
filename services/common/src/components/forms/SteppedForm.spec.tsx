import React from "react";
import { render } from "@testing-library/react";
import SteppedForm from "./SteppedForm";
import Step from "./Step";
import { ReduxWrapper } from "@mds/common/tests/utils/ReduxWrapper";

const props = {
    name: "STEPPED_FORM_NAME",
    tabs: ["basic-information"],
    form: <div />,
    handleTabChange: jest.fn(),
    handleSaveDraft: jest.fn(),
    match: {
        params: {
            tab: "basic-information",
        },
    },
    errors: ["error"],
};


describe("SteppedForm", () => {
    it("renders properly", () => {
        const { container } = render(
            <ReduxWrapper>
                <SteppedForm {...props} activeTab="1">
                    <Step key="1">
                        <div>Step 1 content</div>
                    </Step>
                    <Step key="2">
                        <div>Step 2 content</div>
                    </Step>
                </SteppedForm>
            </ReduxWrapper>
        );
        expect(container).toMatchSnapshot();
    });
});
