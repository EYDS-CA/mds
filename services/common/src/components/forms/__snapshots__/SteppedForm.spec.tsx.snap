// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SteppedForm renders properly 1`] = `
<div>
  <div
    class="ant-row"
  >
    <div
      class="ant-col ant-col-6 stepped-form-menu-container"
    >
      <ul
        class="ant-menu ant-menu-root ant-menu-inline ant-menu-light stepped-form"
        data-menu-list="true"
        role="menu"
        tabindex="0"
      >
        <li
          aria-disabled="false"
          class="ant-menu-item ant-menu-item-selected ant-menu-item-only-child stepped-menu-item"
          data-menu-id="rc-menu-uuid-test-1"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            1
          </span>
        </li>
        <li
          aria-disabled="false"
          class="ant-menu-item ant-menu-item-only-child stepped-menu-item"
          data-menu-id="rc-menu-uuid-test-2"
          role="menuitem"
          style="padding-left: 24px;"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            2
          </span>
        </li>
      </ul>
      <div
        aria-hidden="true"
        style="display: none;"
      />
    </div>
    <div
      class="ant-col ant-col-18"
    >
      <div
        class="stepped-form-form-container"
      >
        <form
          class="ant-form ant-form-vertical common-form common-form-STEPPED_FORM_NAME form-edit"
          id="STEPPED_FORM_NAME"
        >
          <div>
            Step 1 content
          </div>
        </form>
        <div
          class="ant-row ant-row-end stepped-form-button-row"
        >
          <div>
            <button
              class="ant-btn ant-btn-text full-mobile draft-button"
              style="margin-right: 16px;"
              type="button"
            >
              <span>
                Save Draft
              </span>
            </button>
            <button
              class="ant-btn ant-btn-primary"
              type="button"
            >
              <span>
                Save & Continue
              </span>
              <span
                aria-label="right"
                class="anticon anticon-right"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  class=""
                  data-icon="right"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
